cli_tools/diagnostics/main_windows.go

```diff
@@ -168,24 +167,48 @@ func gatherProgramLogs(logs chan logFolder, errs chan error) {
 	logs <- logFolder{"Program", runAll(commands, errs)}
 }
 
-func gatherEventLogs(logs chan logFolder, errs chan error) {
-	dirs := []string{`C:\Windows\System32\winevt\Logs`}
-	paths := make([]string, 0)
-	// Recursively gather all the log files in a directory
-	for len(dirs) > 0 {
-		dir := dirs[0]
-		dirs = dirs[1:]
-		files, _ := ioutil.ReadDir(dir)
-		for _, f := range files {
-			path := filepath.Join(dir, f.Name())
-			if f.IsDir() {
-				dirs = append(dirs, path)
-			} else {
-				paths = append(paths, path)
+func collectFilePaths(roots []string) ([]string, []error) {
+	filePaths := make([]string, 0)
+	errs := make([]error, 0)
+	for _, root := range roots {
+		// Compared filepath.Walk with orginal BFS folder traversal using Measure-Command cmdlet,
+		// looks like almost the same.
+		// 		filepath.Walk -> 4s 973ms
+		// 		original BFS folder traversal -> 4s 897ms
+		// Although filepath.Walk is slower than `find` due to extra lstat calls
+		// https://github.com/golang/go/issues/16399, it should be good enough for this scenario.
+		err := filepath.Walk(root, func(path string, info os.FileInfo, e error) error {
+			if e != nil {
+				return e
+			}
+			if !info.IsDir() {
+				filePaths = append(filePaths, path)
 			}
+			return nil
+		})
+		if err != nil {
+			errs = append(errs, err)
 		}
 	}
-	logs <- logFolder{"Event", paths}
+	return filePaths, errs
+}
+
+func gatherEventLogs(logs chan logFolder, errs chan error) {
+	roots := []string{`C:\Windows\System32\winevt\Logs`}
+	filePaths, ers := collectFilePaths(roots)
+	for _, err := range ers {
+		errs <- err
+	}
+	logs <- logFolder{"Event", filePaths}
+}
+
+func gatherKubernetesLogs(logs chan logFolder, errs chan error) {
+	roots := []string{`C:\etc\kubernetes\logs`, `C:\Windows\MEMORY.dump`}
```

Comment:
The memory dump file could quite large - I found some old ones that I still have around and they're 250-870 MB. Will that be a problem for this tool?

If you need to generate a crash dump for testing purposes you can use the NotMyFault tool from Sysinternals: https://docs.microsoft.com/en-us/sysinternals/downloads/notmyfault

Reply:
Good to know, will test if the tool works fine on big file dump

Reply:
Did some experiments:
1. For 500MB memory dump, after zipped, got 160MB resulting file, taking 3mins to show up on GCS.
2. Faked a 1G memory dump based on 200MB using (cat MEMORY.dmp >> MEMORY.dump), got 140MB resulting file, taking 2mins to show up.
3. Tried to upload 1G file to GCS directly thru gsutil, taking 20s. 
Looks working good.